src/main/java/com/kodilla/ecommercee/service/DbService.java

```diff
@@ -0,0 +1,28 @@
+package com.kodilla.ecommercee.service;
+
+import com.kodilla.ecommercee.controller.GroupNotFoundException;
+import com.kodilla.ecommercee.domain.Group;
+import com.kodilla.ecommercee.repository.GroupRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class DbService {
+    @Autowired
+    private GroupRepository groupRepository;
+
+    public List<Group> getAllGroups() {
+        return groupRepository.findAll();
+    }
+
+    public Optional<Group> getGroupById(final Long id){
+        return groupRepository.findById(id);
+    }
```

Comment:
Ponieważ serwis ma odpowiadać za logikę biznesową, jest to dobre miejsce na weryfikację poprawności zwracanych danych.
Zwróć uwagę, że metoda `findById()` zwraca typ `Optional<Group>`, ponieważ może się zdarzyć, że w bazie nie ma grupy o podanym id.
Wydaje mi się, że właśnie serwis jest odpowiednim miejscem na rzucenie w takim wypadku wyjątku.
Dlatego zmieniłbym ją na coś takiego:
```
    public Group getGroupById(final Long id){
        return groupRepository.findById(id).orElseThrow(
                () -> new RuntimeException("Group with id=" + id + " doesn't exist.")
        );
    }
```